auacm/app/modules/submission_manager/judge.py

```diff
@@ -50,37 +46,39 @@ def allowed_filetype(filename):
     return ('.' in filename and
             filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS)
 
+
 class Judge:
     """A container for a judging instance. It contains the submission, the
     source code that goes along with the submission, and the time limit of the
     problem.
     """
 
-    def __init__(self, submission, uploaded_file, time_limit):
+    def __init__(self, pid, submission_path, uploaded_file, time_limit, on_status=None):
```

Comment:
For now, no, there's not anything else that we'd be doing, but it decouples the logic of publishing the status to clients from the judge to the calling function. I think it's cleaner that way. It means that the judge can be run as more of a black box. All logic that actually interacts with the network can be maintained independent of the judge itself.